internal/config/config.go

```diff
@@ -200,6 +200,48 @@ func (c *Config) NewFabricClient() *v4.APIClient {
 	return client
 }
 
+// NewFabricClientForSDK returns a terraform sdkv2 plugin compatible
+// equinix-sdk-go/fabricv4 client to be used to access Fabric's V4 APIs
+func (c *Config) NewFabricClientForSDK(d *schema.ResourceData) *fabricv4.APIClient {
```

Comment:
Worth considering #601 here as well so you don't have to explicitly wire up the auth token in every CRUD method

Reply:
Also work taking a look at #617 while you are here

Reply:
@displague , the user agent is taken care of when switching over to equinix-sdk-go/fabricv4. That's included in this PR.

@ctreatma , have you tested #601? I tried testing it out with the change that I've made here and getting timeout exceeded while waiting for OAUTH token. I could've made an error in my attempt to make the same change with the sdk client creation method instead of just setting it during config.